process_report/process_report.py

```diff
@@ -46,43 +55,53 @@ def load_institute_map() -> dict:
     return institute_map
 
 
-def load_old_pis(old_pi_file):
-    old_pi_dict = dict()
-
+def load_old_pis(old_pi_file) -> pandas.DataFrame:
     try:
-        with open(old_pi_file) as f:
-            for pi_info in f:
-                pi, first_month = pi_info.strip().split(",")
-                old_pi_dict[pi] = first_month
+        old_pi_df = pandas.read_csv(old_pi_file)
     except FileNotFoundError:
         print("Applying credit 0002 failed. Old PI file does not exist")
         sys.exit(1)
 
-    return old_pi_dict
+    return old_pi_df
 
 
-def dump_old_pis(old_pi_file, old_pi_dict: dict):
-    with open(old_pi_file, "w") as f:
-        for pi, first_month in old_pi_dict.items():
-            f.write(f"{pi},{first_month}\n")
+def dump_old_pis(old_pi_file, old_pi_df: pandas.DataFrame):
+    old_pi_df.to_csv(old_pi_file, index=False)
 
 
-def is_old_pi(old_pi_dict, pi, invoice_month):
-    first_invoice_month = old_pi_dict.get(pi, invoice_month)
-    if compare_invoice_month(first_invoice_month, invoice_month):
+def get_pi_age(old_pi_df: pandas.DataFrame, pi, invoice_month):
+    """Returns time difference between current invoice month and PI's first invoice month
+    I.e 0 for new PIs
+    Can be negative if processing past invoices"""
```

Comment:
but it raises an error when it's negative

Reply:
I should have wrote the docstring a bit clearer. When writing this, I believed it should be intended behavior for the program to raise an error if the program was processing a past invoice, but encountered a PI that was suppose to appear in the future. 

I.e if the PI file had an entry for PI A for 2024-01, but PI A was found when processing past invoices, an error should be raised since that might suggest the invoice was wrong, or the program was faulty in the past.

I will remove the ` Can be negative if processing past invoices` comment from the docstring